src/validation.cpp

```diff
@@ -2624,7 +2624,7 @@ bool CChainState::DisconnectTip(BlockValidationState& state, DisconnectedBlockTr
         }
     }
 
-    m_chain.SetTip(pindexDelete->pprev);
+    m_chain.SetTip(*Assert(pindexDelete->pprev));
```

Comment:
In commit "Remove unused SetTip(nullptr) code" (faf57a6898590cecfbb918d73abbf2c100206b8e)

I think it would be good to add an assert to the top of the function to make it more obvious it will crash if called on the genesis block:

```diff
diff --git a/src/validation.cpp b/src/validation.cpp
index da18fec1d85..75d06841e3f 100644
--- a/src/validation.cpp
+++ b/src/validation.cpp
@@ -2577,6 +2577,7 @@ bool CChainState::DisconnectTip(BlockValidationState& state, DisconnectedBlockTr
 
     CBlockIndex *pindexDelete = m_chain.Tip();
     assert(pindexDelete);
+    assert(pindexDelete->pprev);
     // Read block from disk.
     std::shared_ptr<CBlock> pblock = std::make_shared<CBlock>();
     CBlock& block = *pblock;

```

Reply:
Nice, done.